### PR TITLE
[WIP] Get JabRef running in eclipse

### PR DESCRIPTION
**Pull request by Siedlerchr**
I finally figured out how to make JabRef working with java 11 in eclipse.
Requires eclipse 2019-06 minimum

uncomment java version checker, does not work inside eclipse
TODO: Write up or generate the classpath entries with add opens as well
TODO: mark jmh as test resource folder

1. Run gradlew eclipse
2. Go to build path  -> Libraries
3.  Move all test related libs to classpath
4. Move jmh to classpath as well
5. Go to Source -> Mark src/jmh as test folder
6. Go to Module dependencies 
6.1. Select `org.controlsfx.controls` in the list
6.2. Add export and opens entry  for `impl.org.controlsfx.autocompletion`, `impl.org.controlsfx.skin` and `org.controlsfx.control.textfield`  to `org.jabref`
6.3. Select `javafx.controls` add epxorts and opens entry for `com.sun.javafx.scene.control`  to `org.jabref`

JPMS options should look like this:
```
--add-exports org.controlsfx.controls/impl.org.controlsfx.autocompletion=org.jabref --add-exports org.controlsfx.controls/impl.org.controlsfx.skin=org.jabref --add-exports org.controlsfx.controls/org.controlsfx.control.textfield=org.jabref --add-opens org.controlsfx.controls/impl.org.controlsfx.autocompletion=org.jabref --add-opens org.controlsfx.controls/impl.org.controlsfx.skin=org.jabref --add-opens org.controlsfx.controls/org.controlsfx.control.textfield=org.jabref --add-exports javafx.controls/com.sun.javafx.scene.control=org.jabref --add-opens javafx.controls/com.sun.javafx.scene.control=org.jabref
```

7. Create a run/debug configuration for main class `org.jabref.JabRefLauncher` 
8. In the arguments tab enter this for VM arguments.
```
-Djabref.theme.css --patch-module test=fastparse_2.12-1.0.0.jar --patch-module test2=fastparse-utils_2.12-1.0.0.jar --patch-module test3=sourcecode_2.12-0.1.4.jar --patch-module org.jabref=build\resources\main --add-exports javafx.controls/com.sun.javafx.scene.control=org.jabref --add-exports org.controlsfx.controls/impl.org.controlsfx.skin=org.jabref --add-opens javafx.controls/javafx.scene.control=org.jabref --add-opens org.controlsfx.controls/org.controlsfx.control.textfield=org.jabref --add-exports javafx.graphics/com.sun.javafx.scene=org.controlsfx.controls --add-exports javafx.graphics/com.sun.javafx.scene.traversal=org.controlsfx.controls --add-exports javafx.graphics/com.sun.javafx.css=org.controlsfx.controls --add-exports javafx.controls/com.sun.javafx.scene.control.behavior=org.controlsfx.controls --add-exports javafx.controls/com.sun.javafx.scene.control=org.controlsfx.controls --add-exports javafx.controls/com.sun.javafx.scene.control.inputmap=org.controlsfx.controls --add-exports javafx.base/com.sun.javafx.event=org.controlsfx.controls --add-exports javafx.base/com.sun.javafx.collections=org.controlsfx.controls --add-exports javafx.base/com.sun.javafx.runtime=org.controlsfx.controls --add-exports javafx.web/com.sun.webkit=org.controlsfx.controls --add-exports javafx.graphics/com.sun.javafx.css=org.controlsfx.controls --add-opens javafx.controls/javafx.scene.control.skin=org.controlsfx.controls --add-opens javafx.graphics/javafx.scene=org.controlsfx.controls"
```

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,3 @@
-[submodule "src/main/resources/csl-styles"]
```
Comment (tobiasdiez) [inline .gitmodules]: What's the reason for this change?

File: src/main/java/org/jabref/JabRefMain.java [hunk 2]
```diff
@@ -48,7 +48,7 @@ public static void main(String[] args) {
     public void start(Stage mainStage) throws Exception {
         try {
             // Fail on unsupported Java versions
-            ensureCorrectJavaVersion();
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/JabRefMain.java]: Could you please re-enable this? Did it work before in eclipse?

File: .gitmodules (hunk 1 above)
Comment (Siedlerchr) [reply .gitmodules]: Hmm this was not intended

**Comment by tobiasdiez** [comment]
I have reverted the merge for the moment.

**Comment by Siedlerchr** [comment]
thanks I will investitage further.
